app/models/queued_email/author_request.rb

```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+# Request to be the author of an object
+class QueuedEmail::AuthorRequest < QueuedEmail
+  def obj_id
+    get_integer(:obj_id)
+  end
+
+  def obj_type
+    get_string(:obj_type)
+  end
+
+  def subject
+    get_string(:subject)
+  end
+
+  def content
+    get_note
+  end
+
+  def self.create_email(sender, recipient, object, subject, message)
+    result = create(sender, recipient)
+    raise("Missing object!") unless object
+
+    result.add_integer(:obj_id, object.id)
+    result.add_string(:obj_type, object.type_tag.to_s)
+    result.add_string(:subject, subject)
+    result.set_note(message)
+    result.finish
+    result
```

Comment:
Oooh!  I just thought of something.  What if QueuedEmail#add_xxx, set_note, finish all returned self?  Then these could all be chained:

create().add_xxx(xxx).add_xxx(xxx).set_note(xxx).finish

That would be so slick! :)

Reply:
The `return unless object` currently interrupts that chain, should it come before?

Reply:
Trying it.  Like so?
```
    raise("Missing object!") unless object

    create(sender, recipient).
      add_integer(:obj_id, object.id).
      add_string(:obj_type, object.type_tag.to_s).
      add_string(:subject, subject).
      set_note(message).finish
```